buffer/src/main/java/io/netty/buffer/AbstractByteBuf.java

```diff
@@ -267,6 +267,7 @@ public ByteBuf ensureWritable(int minWritableBytes) {
     }
 
     private void ensureWritable0(int minWritableBytes) {
+        ensureAccessible();
```

Comment:
I was looking at `public int ensureWritable(int minWritableBytes, boolean force)` but it even with the modifier as protected it still feels backwards.